scene/3d/sprite_3d.cpp

```diff
@@ -1112,16 +1112,13 @@ void AnimatedSprite3D::set_sprite_frames(const Ref<SpriteFrames> &p_frames) {
 
 		List<StringName> al;
 		frames->get_animation_list(&al);
-		if (al.size() == 0) {
-			set_animation(StringName());
+
+		animation = StringName();
+		if (!al.is_empty()) {
+			set_animation(frames->has_animation(animation) ? animation : al[0]);
```

Comment:
```suggestion
			set_animation(frames->has_animation(animation) ? animation : al.front());
```

Reply:
I tried it, it turned out I needed `al.front()->get()` instead, which I find a lot less clear and the same amount of scary since it requires a ptr deref, so i would prefer to keep `al[0]`

Reply:
You can instead do:
```suggestion
			set_animation(frames->has_animation(animation) ? animation : *al.front());
```
Which should be clearer, but it's either way 🙂